backend/views/syson-tree-explorer-view/src/main/java/org/eclipse/syson/tree/explorer/view/SysONDuplicateTargetBrowserTreeDescriptionIdProviderDelegate.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.syson.tree.explorer.view;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.eclipse.sirius.components.collaborative.browser.api.IModelBrowserTreeDescriptionIdProviderDelegate;
+import org.eclipse.sirius.components.core.URLParser;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IObjectSearchService;
+import org.eclipse.syson.sysml.Element;
+import org.eclipse.syson.sysml.Relationship;
+import org.springframework.stereotype.Service;
+
+/**
+ * {@link IModelBrowserTreeDescriptionIdProviderDelegate} that provides a custom tree to select the future owner of a duplication.
+ *
+ * @author Arthur Daussy
+ */
+@Service
+public class SysONDuplicateTargetBrowserTreeDescriptionIdProviderDelegate implements IModelBrowserTreeDescriptionIdProviderDelegate {
+
+    private final URLParser urlParser;
+
+    private final IObjectSearchService objectSearchService;
+
+    public SysONDuplicateTargetBrowserTreeDescriptionIdProviderDelegate(URLParser urlParser, IObjectSearchService objectSearchService) {
+        this.urlParser = urlParser;
```

Comment:
= urlParser; => = Objects.requireNonNull(urlParser);

here, the one below and everywhere else in this PR and your future PRs.

Reply:
Done.